core/consensus/babe/impl/babe_lottery_impl.cpp

```diff
@@ -42,8 +42,8 @@ namespace kagome::consensus {
         epoch.randomness.begin(), epoch.randomness.end(), vrf_input.begin());
 
     auto slot_number_begin = vrf_input.begin() + vrf_constants::OUTPUT_SIZE;
-    for (BabeSlotNumber i = epoch.epoch_index * epoch.epoch_duration;
-         i < (epoch.epoch_index + 1) * epoch.epoch_duration;
+    for (BabeSlotNumber i = epoch.start_slot;
+         i < epoch.start_slot + epoch.epoch_duration;
```

Comment:
```epoch_duration``` I'm not sure that it is a good name for this variable. It's not convenient, that ```epoch_duration``` is in slots but not in time units.

Reply:
Moreover ```slot_duration``` is a time.